File: 프로그래머스/1/42840. 모의고사/모의고사.java
import java.util.*;

class Solution {
    public List solution(int[] answers) {
        List<Integer> answer = new ArrayList<>();
        int[] scores = new int[3];
        int[] a = {1, 2, 3, 4, 5};
        int[] b = {2, 1, 2, 3, 2, 4, 2, 5};
        int[] c = {3, 3, 1, 1, 2, 2, 4, 4, 5, 5};
        
        for (int i=0; i<answers.length; i++) {
            if (answers[i] == a[i%a.length])
                    scores[0]++;
            if (answers[i] == b[i%b.length])
                    scores[1]++;
            if (answers[i] == c[i%c.length])
                    scores[2]++;
        }
        
        int maxScore = Math.max(scores[0], Math.max(scores[1], scores[2]));
        for (int i=0; i<scores.length; i++){
            if (scores[i] == maxScore)
                answer.add(i+1);
        }
        return answer;
    }
}

File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    correct = []
    cnt = [0, 0, 0]
    one, two, three = [1, 2, 3, 4, 5], [2, 1, 2, 3, 2, 4, 2, 5], [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    
    for idx, answer in enumerate(answers):
        if one[idx % 5] == answer:
            cnt[0] += 1
        if two[idx % 8] == answer:
            cnt[1] += 1
        if three[idx % 10] == answer:
            cnt[2] += 1
            
    maxN = max(cnt)
    for i in range(3):
        if maxN == cnt[i]:
            correct.append(i+1)
    return correct